app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -302,83 +209,113 @@ class List extends Component {
+    };
 
-        return {
+    renderSectionHeader = ({section}) => {
+        const {intl, styles} = this.props;
+        const {
+            action,
+            bottomSeparator,
+            defaultMessage,
             id,
-            isTitle: true,
-            title: (
-                <View>
-                    {topDivider && this.renderDivider(styles, 0)}
-                    <View style={styles.titleContainer}>
-                        <Text style={styles.title}>
-                            {formatMessage({id, defaultMessage}).toUpperCase()}
-                        </Text>
-                        {action && this.renderSectionAction(styles, action)}
-                    </View>
-                    {bottomDivider && this.renderDivider(styles, 0)}
+            topSeparator
+        } = section;
+
+        return (
+            <View>
+                {topSeparator && this.renderSectionSeparator()}
+                <View style={styles.titleContainer}>
+                    <Text style={styles.title}>
+                        {intl.formatMessage({id, defaultMessage}).toUpperCase()}
+                    </Text>
+                    {action && this.renderSectionAction(styles, action)}
                 </View>
-            )
-        };
+                {bottomSeparator && this.renderSectionSeparator()}
+            </View>
+        );
     };
 
-    render() {
-        const {styles} = this.props;
+    scrollToTop = () => {
+        if (this.refs.list) {
+            this.refs.list.scrollToOffset({
+                x: 0,
+                y: 0,
+                animated: true
+            });
+        }
+    };
 
-        const {dataSource, showAbove} = this.state;
+    emitUnreadIndicatorChange = debounce((show) => {
+        EventEmitter.emit(SHOULD_SHOW_EVENT, show);
+    }, 100);
 
-        let above;
-        if (showAbove) {
-            above = (
-                <UnreadIndicator
-                    style={[styles.above, {width: (this.width - 40)}]}
-                    onPress={this.scrollToTop}
-                    text={(
-                        <FormattedText
-                            style={styles.indicatorText}
-                            id='sidebar.unreadAbove'
-                            defaultMessage='Unread post(s) above'
-                        />
-                    )}
-                />
-            );
+    updateUnreadIndicators = ({viewableItems}) => {
+        const {unreadChannelIds} = this.props;
+        const firstUnread = unreadChannelIds[0];
+        if (firstUnread) {
+            const isVisible = viewableItems.find((v) => v.item === firstUnread);
+
+            if (isVisible) {
+                return this.emitUnreadIndicatorChange(false);
+            }
+            return this.emitUnreadIndicatorChange(true);
         }
 
+        return this.emitUnreadIndicatorChange(false);
+    };
+
+    render() {
+        const {styles, theme} = this.props;
+        const sections = this.buildSections();
```

Comment:
Should this be stored in state?

Reply:
then you'll get complains about `Avoidable updates` when this should be created just once and you already have the data in the props for it

Reply:
Isn't it still updating if we do it this way though? Maybe it's not causing that warning, but it's giving new props to the SectionList which'll cause it to rerender since it's a PureComponent